azure-core/setup.py

```diff
@@ -79,6 +79,7 @@
     install_requires=[
         "requests>=2.18.4",
         "certifi>=2017.4.17",
+        "typing-extensions>=3.7.2",
```

Comment:
If at all possible, I'd like to see if we can avoid installing typing-extensions as part of the "real" setup. Conditional imports when TYPING is set to True?

Reply:
I learned a lot thinking through the consequences of doing this.